hudi-common/src/main/java/org/apache/hudi/common/table/TableSchemaResolver.java

```diff
@@ -165,7 +165,7 @@ public Schema getTableAvroSchema(boolean includeMetadataFields) throws Exception
       if (includeMetadataFields) {
         return HoodieAvroUtils.addMetadataFields(schemaFromTableConfig.get(), hasOperationField);
       } else {
-        return schemaFromTableConfig.get();
+        return HoodieAvroUtils.removeMetadataFields(schemaFromTableConfig.get());
```

Comment:
why is this changed?

Reply:
Hello, When create schema already contains metadata fields but includeMetadataField is set to false, can avro schema that does not contain metadata fields be returned? Considering the above scenario, I modified the code on line 168.

Reply:
I answered your questions in the comments and hope you can review my code again.

Reply:
I think it's inconsiderate and I've deleted it.